src/Player/PlayerStrategies.cpp

```diff
@@ -0,0 +1,613 @@
+            weakest_country->set_armies(total_armies_divided_by_two);
+        } else {
+            strongest_neighbor->set_armies(total_armies_divided_by_two+1);
+            weakest_country->set_armies(total_armies_divided_by_two);
+        }
+
+        answer.emplace_back(strongest_neighbor); // source
+        answer.emplace_back(weakest_country); //target
+        return answer;
+    }
+
+    void BenevolentPlayerStrategy::reinforce(int i) {
+        Terminal::debug("Performing reinforce from a benevolent player" + player->get_color());
+
+        vector<Board::Country *> owned_countries = player->get_countries();
+        //Find weakest country
+        int smallest_army = 99999;
+        Country * weakest_country = nullptr;
+        for (auto &country : owned_countries){
+            if (country->get_armies() < smallest_army){
+                weakest_country = country;
+                smallest_army = country->get_armies();
+            }
+        }
+
+        if(weakest_country == nullptr) {
+            Terminal::error("Weakest country is null, this should NEVER happen");
+            return;
+        }
+
+        weakest_country->set_armies(weakest_country->get_armies() + i);
+
+    }
+
+    BenevolentPlayerStrategy::~BenevolentPlayerStrategy() {
+
+    }
+
+    /**
+ * an aggressive computer player that focuses on attack (reinforces its strongest
+country, then always attack with it until it cannot attack anymore, then fortifies in order to maximize aggregation of
+forces in one country),
+ * @param i
+ */
+
+    bool AggressivePlayerStrategy::attack() {
+        bool has_won_a_battle = false;
+        //Find strongest country
+        int biggest_army = 0;
+        Country * strongest_country = nullptr;
+        for (auto &country : player->get_countries()){
+            if (country->get_armies() > biggest_army){
+                // Verify if the country has a neighbor that player owns and has army
+                for (auto &neighbor_country : * country->get_neighbors()){
+                    if (neighbor_country->get_owner() == player && neighbor_country->get_armies() > 1){
+                        strongest_country = country;
+                        biggest_army = country->get_armies();
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (strongest_country == nullptr){
+            Terminal::error("Strongest_country is null, this should NEVER happen");
+            return has_won_a_battle;
+        }
+
+        while (strongest_country->get_armies() > 1){
+            Board::Country * country_under_attack = nullptr;
+            for (auto &neighbor : *strongest_country->get_neighbors()){
+                if (neighbor->get_owner() != this->player) {
+                    country_under_attack = neighbor;
+                    break;
+                }
+            }
+            if (country_under_attack == nullptr) return has_won_a_battle;
```

Comment:
Does this mean the AI won't attack if his strongest country is surrounded by countries it owns?
Should he the select another country on his borders? Not sure here...

Reply:
The strongest country will attack the first country it sees which isn't his own. `country_under_attack` is the target of the attack